### PR TITLE
8323794: Remove unused jimage compressor plugin configuration

### PR DESCRIPTION
**Pull request by cl4es**
There's an unused concept of a pluginConfig that is passed into the jlink compress plugins, however we always pass null here and the code seems broken (the pluginConfig wouldn't have been stored properly). This seem to be a left-over from a more generic plugin design that never came to fruition.

This PR cleans out this code from the plugins and decompressors, while keeping the compressed header format intact for backwards compatibility (and in case we want to revisit this in the future)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8323794](https://bugs.openjdk.org/browse/JDK-8323794): Remove unused jimage compressor plugin configuration (**Enhancement** - P4)

### Reviewers
 * [Jim Laskey](https://openjdk.org/census#jlaskey) (@JimLaskey - **Reviewer**) ⚠️ Review applies to [7df80e39](https://git.openjdk.org/jdk/pull/17443/files/7df80e396080a4c6e6cc495c7a74f19e192191c2)
 * [Mandy Chung](https://openjdk.org/census#mchung) (@mlchung - **Reviewer**) ⚠️ Review applies to [a65ffbf5](https://git.openjdk.org/jdk/pull/17443/files/a65ffbf5a22d59a56059cd9f022fd147ca692284)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17443/head:pull/17443` \
`$ git checkout pull/17443`

Update a local copy of the PR: \
`$ git checkout pull/17443` \
`$ git pull https://git.openjdk.org/jdk.git pull/17443/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17443`

View PR using the GUI difftool: \
`$ git pr show -t 17443`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17443.diff">https://git.openjdk.org/jdk/pull/17443.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17443#issuecomment-1893516292)

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/jimage/decompressor/CompressedResourceHeader.java [hunk 1]
```diff
@@ -115,16 +101,16 @@ public static CompressedResourceHeader readFromResource(ByteOrder order,
         }
         ByteBuffer buffer = ByteBuffer.wrap(resource, 0, SIZE);
         buffer.order(order);
-        int magic = buffer.getInt();
+        int magic = buffer.getInt(0);
```
Comment (JimLaskey) [inline src/java.base/share/classes/jdk/internal/jimage/decompressor/CompressedResourceHeader.java:104]: Named constant for offsets?

File: src/java.base/share/classes/jdk/internal/jimage/decompressor/CompressedResourceHeader.java [hunk 2]
```diff
@@ -115,16 +101,16 @@ public static CompressedResourceHeader readFromResource(ByteOrder order,
...
         if(magic != MAGIC) {
             return null;
         }
-        long size = buffer.getLong();
-        long uncompressedSize = buffer.getLong();
-        int decompressorNameOffset = buffer.getInt();
-        int contentIndex = buffer.getInt();
-        byte isTerminal = buffer.get();
+        long size = buffer.getLong(4);
```
Comment (JimLaskey) [inline src/java.base/share/classes/jdk/internal/jimage/decompressor/CompressedResourceHeader.java:108]: Named constant for offsets?

**Review by JimLaskey — CHANGES_REQUESTED** [review]

**Review by JimLaskey — APPROVED** [review]
LGTM

**Review by mlchung — APPROVED** [review]
Looks good.

**Review by mlchung — APPROVED** [review]

**Comment by cl4es** [comment]
Thanks for reviewing and re-reviewing! I caught and fixed a few issues in testing (tier1-4), extending pre-integration testing to tier5 and taking another manual pass over all related tests before integrating.

**Pull request closed without merging**